gravitee-apim-console-webui/src/management/api/documentation-v4/documentation-pages-tab/api-documentation-v4-documentation-pages-tab.component.ts

```diff
@@ -148,6 +169,40 @@ export class ApiDocumentationV4DocumentationPagesTabComponent implements OnInit,
     });
   }
 
+  generate() {
+    this.matDialog
+      .open<ApiDocumentationV4NewtAiDialogComponent, ApiDocumentationV4NewtAiDialogData>(ApiDocumentationV4NewtAiDialogComponent, {
+        width: GIO_DIALOG_WIDTH.MEDIUM,
+        data: {
+          state: this.specGenState,
+        } as ApiDocumentationV4NewtAiDialogData,
+      })
+      .afterClosed()
+      .pipe(
+        filter((start) => !!start),
+        switchMap(() => this.apiSpecGenService.postJob(this.api.id)),
```

Comment:
```suggestion
        switchMap(() => this.apiSpecGenService.postJob(this.api.id)),
        takeUntil(this.unsubscribe$)
```
in general, each subscriber must declare an unsubscribe if the page is destroyed

Reply:
Done